File: Longest Substring with At Least K Repeating Characters/Longest_Substring_with_At_Least_K_Repeating_Characters.cpp
class Solution {
public:
    int longestSubstring(string s, int k) {
        int n = s.size();
        int ans = 0;
        for(int i = 0; i<n; i++) {
            vector<int> fre(26,0);
            for(int j = i; j<n; j++) {
                fre[s[j]-'a']++;
                bool flag = true;
                for(int x = 0; x<26; x++) {
                    if(fre[x] > 0) {
                        if(fre[x] < k) {
                            flag = false;
                            break;
                        }
                    }
                }
                if(flag) {
                    ans = max(ans,j-i+1);
                }
            }
            
        }
        return ans;
    }
};

File: Longest Substring with At Least K Repeating Characters/Longest_Substring_with_At_Least_K_Repeating_Characters.py
class Solution:
    def longestSubstring(self, s: str, k: int) -> int:
        n = len(s)
        ans = 0
        for i in range(0,n):
            fre = [0]*26
            for j in range(i,n):
                fre[ord(s[j])-ord('a')] +=1
                flag = True
                for x in range(0,26):
                    if fre[x] > 0:
                        if fre[x] < k:
                            flag = False
                            break
                if flag:
                    ans = max(ans,j-i+1)
        return ans
        